plugins/ip.rb

```diff
@@ -32,9 +18,8 @@ def self.pre_validate options
 
     # if we don't have a gateway address we'll compute one
     if ! options[:gateway]
```

Comment:
You technically don' t even need this conditional since you've removed the argument to set this.